sdk/core/core-http/lib/serviceClient.ts

```diff
@@ -593,6 +593,86 @@ function createDefaultRequestPolicyFactories(
   return factories;
 }
 
+export function createPipelineFromOptions(
```

Comment:
This is a separate pathway to be used by our convenience layer clients, the original default pipeline creation logic still applies for generated ServiceClient implementations.

Reply:
This is such a handy function!!! 😍

Reply:
Should this also be adding the logging policy?

Reply:
By "original default pipeline creation logic" , do you mean by using the `ServiceClientOptions`?

Reply:
Yep, `ServiceClientOptions` as a parameter to the `ServiceClient` constructor.

Reply:
@bterlson Yep, it should also configure `logPolicy`.  I'm starting to see the need for a separate interface (or an `extends` interface) that is for internal SDK configuration to add things like the allowed list of headers for logging, etc.  Basically, anything that shouldn't be configured by a user of the SDK.